xla/stream_executor/device_description.h

```diff
@@ -69,98 +104,104 @@ class RocmComputeCapability {
+  bool gfx12_discrete() const { return IsThisGfxInAnyList(kGfx12Discrete); }
 
   bool has_nhwc_layout_support() const { return gfx9_mi100_or_later(); }
 
-  bool has_bf16_dtype_support() const { return gfx9_mi100_or_later(); }
+  bool has_bf16_dtype_support() const {
+    return gfx9_mi100_or_later() || gfx12() || gfx11();
+  }
 
   bool has_fast_fp16_support() const {
-    return gfx9_mi100_or_later() || gfx10_rx68xx() || gfx10_rx69xx() || gfx11();
+    return gfx9_mi100_or_later() || gfx11() || gfx10_rx68xx() || gfx10_rx69xx();
   }
 
   bool has_mfma_instr_support() const { return gfx9_mi100_or_later(); }
 
   bool has_amd_matrix_core() const {
-    return (gfx9_mi100_or_later() || gfx_version().find("gfx11") ||
-            gfx_version().find("gfx12"));
+    return gfx9_mi100_or_later() || gfx12() || gfx11();
   }
 
   bool has_packed_fp16_atomics_support() const { return gfx9_mi100_or_later(); }
 
   bool has_packed_bf16_atomics_support() const { return gfx9_mi300_series(); }
 
   bool fence_before_barrier() const {
-    return gfx_version() != "gfx900" && gfx_version() != "gfx906";
+    static constexpr absl::string_view kList[] = {"gfx900", "gfx906"};
+    return !IsThisGfxInAnyList(kList);
   }
 
   bool has_hipblaslt() const {
-    return gfx9_mi200_or_later() || gfx1200() || gfx1201();
+    return IsThisGfxInAnyList(kMI300Series, kMI200Series, kGfx12Discrete,
+                              kGfx11Discrete, kGfx11Apu);
   }
 
   bool has_fp8_support() const {
     return has_ocp_fp8_support() || has_nanoo_fp8_support();
   }
 
-  bool has_ocp_fp8_support() const {
-    return gfx1200() || gfx1201() || gfx9_mi350();
-  }
+  bool has_ocp_fp8_support() const { return gfx9_mi350() || gfx12_discrete(); }
 
   bool has_nanoo_fp8_support() const { return gfx9_mi300(); }
 
-  std::string ToString() const { return gcn_arch_name(); }
-
-  RocmComputeCapabilityProto ToProto() const {
-    RocmComputeCapabilityProto proto;
-    proto.set_gcn_arch_name(gcn_arch_name_);
-    return proto;
+ private:
+  /// \brief Takes one or more arrays of string-like objects and tests if the
+  /// result of `gfx_version()` matches to any string in any of the arrays.
+  template <typename... ArrayOfStrings>
+  bool IsThisGfxInAnyList(ArrayOfStrings &&...arr) const {
+    static_assert(sizeof...(arr) >= 1);
+    const auto gfx = gfx_version();
+    return (implIsThisGfxInAnyList(std::begin(arr), std::end(arr), gfx) || ...);
   }
 
-  bool operator==(const RocmComputeCapability &other) const {
-    return gcn_arch_name_ == other.gcn_arch_name_;
+  /// \brief Template-less implementation of IsThisGfxInAnyList().
+  /// \warning Don't use directly!
+  bool implIsThisGfxInAnyList(const absl::string_view *beg,
+                              const absl::string_view *end,
+                              const std::string &gfx) const {
+    return std::any_of(beg, end, [&gfx = gfx](const absl::string_view &s) {
```

Comment:
Nit: I think it would be more idiomatic to just use `&` instead of `&gfx = gfx`.